### ISSUE TITLE
Improper Bounding box 

### ISSUE THREAD

**Issue by ngaurav07** [nwojke/deep_sort#223]
I am applying deep sort in my college project with yolov4 . Everything is right except in some videos bounding boxes are large and wide than YOLO. YOLO has good bounding box outcome but deep sort has output of larger boxes. Hope to get answer soon. Thankyou in advance

**Comment by spranjal25** [comment]
> I am applying deep sort in my college project with yolov4 . Everything is right except in some videos bounding boxes are large and wide than YOLO. YOLO has good bounding box outcome but deep sort has output of larger boxes. Hope to get answer soon. Thankyou in advance

Hi @gaulove , I am also trying to implement something similar would you be able to give me some idea on how did you integrate your detection model with this tracker?.
Thanks in advance!

**Comment by akashAD98** [comment]
@spranjal25 @gaulove i am also facing the same issue [https://github.com/guhuozhengling/deepsort_yolov4/issues/6](url) please let me know if you get solution.

**Comment by spranjal25** [comment]
> @spranjal25 @gaulove i am also facing the same issue [https://github.com/guhuozhengling/deepsort_yolov4/issues/6](url) please let me know if you get solution.

hi @akashAD98 , I solved the issue. you should check the bounding boxes that you might be returning to the DeepSORT model follow a particular format. For my case I didn't use this implementation of DeepSORT. However, if you look at the basic procedure, your detector passes the bounding box detections to the tracker and then the tracker keeps updating those boxes for the mentioned frames, The reasons why your boxes are abnormally placed can be:

1. The coordinates are normalized and they are not multiplied with the image dimensions while plotting on the frame (opencv functions) 
2.  the other reason, very unlikely to be though, can be a problem with your weights.

Can you tell me what procedure you followed exactly?

**Comment by akashAD98** [comment]
@spranjal25  I was using pre-trained weights of yolov4-CSP & yolov4-mish, I was successfully detected with yolov4.weights. but for this csp& mish, I'm getting a big bounding box.I think due to a change in the structure of yolov4-csp=mish maybe this is problem.
because bbox calculation is different in scaled yolo . can you please ping me on aksdesai1998@gmail.com  so we will discuss more on this.Thanks

**Comment by akashAD98** [comment]
@spranjal25  I was using pre-trained weights of yolov4-CSP & yolov4-mish, I was successfully detected with yolov4.weights. but for this csp& mish, I'm getting a big bounding box.I think due to a change in the structure of yolov4-csp=mish maybe this is problem.
because bbox calculation is different in scaled yolo . can you please ping me on aksdesai1998@gmail.com  so we will discuss more on this.Thanks

**Comment by gaulove** [comment]
could you send the output ?

**Comment by akashAD98** [comment]
@ngaurav07 hey can you share the script of object tracking? Thanks in advance.

**Comment by ngaurav07** [comment]
Sorry, Bro, I have left that company and cant share it. I can help you in case of any problems.

**Comment by akashAD98** [comment]
@ngaurav07 have you converted yolov4-mish weight into tensorflow saved model format? or are you directly using yolo weight for tracking object?

**Comment by hsiaochinglin** [comment]
![1697778609912](https://github.com/nwojke/deep_sort/assets/42636139/ea53677e-1917-4ba0-9e1d-9f2880f40b9c)
Hi, I have a similar problem about bounding box. I can run the video but my output video is like the picture i paste. Can anyone give me a suggestion? thanks a lot!

**Comment by batiamida** [comment]
I believe this is due to calculating aspect ratio as a patch image not taking into the account aspect ratio of predicted bbox.
